test/IRGen/type_layout_reference_storage_objc.swift

```diff
@@ -23,6 +24,16 @@ struct ReferenceStorageTypeLayout<T> {
   // CHECK: store i8** getelementptr inbounds (i8*, i8** @_T0BOSgXwWV, i32 17)
   weak            var cwi: C!
 
+  // -- ObjC-refcounted archetype
+  // CHECK: store i8** getelementptr inbounds (i8*, i8** @_T0BOXoWV, i32 17)
+  unowned(safe)   var os:  ObjC
+  // CHECK: store i8** getelementptr inbounds (i8*, i8** @_T0BomWV, i32 17)
```

Comment:
Hey @jckarter and @rjmccall, why do we lower unowned(unsafe) references as "NativeObject.Type"? Does this have the right spare bits and extra inhabitants given that unowned(unsafe) can contain Objective-C tagged pointers?